pkg/storage/bucket/config/http.go

```diff
@@ -0,0 +1,35 @@
+package config
```

Comment:
Small thing, but I got a little confused over the naming of this package.
It doesn't contain all bucket config.  I think it exists to avoid a circular dependency?
Given what it does, I might call the package `http` and the struct `Config`.

Reply:
Yup it was to avoid the circular dependency. Alternatively we could make `azure` depend on `s3.HTTPConfig`, but that felt worse, also `s3.HTTPConfig` has a `Transport` that would be unused by `azure`.

I've renamed `config.HTTP` -> `http.Config` in the latest commit.